### ISSUE TITLE
Fixed columns are  disappeared in antd table when horizontal scroll moved

### ISSUE THREAD

**Issue by sb-lee** [ant-design/ant-design#22361]
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/fixed-columns-and-header-ant-design-demo-7sgyn)

### Steps to reproduce
1. modify table option at `tableLayout="fixed" , rowSelection={{ fixed: true }}, pagination={false},    scroll={{ x: 1500 }}`  in https://codesandbox.io/s/16d8b 

### What is expected?
Fixed columns are fixed

### What is actually happening?
Fixed columns are disappeared with row selection

| Environment | Info |
|---|---|
| antd | 4.0.3 |
| React | 16.13 |
| System | window 10  (1909) |
| Browser | chrome latest |

**Comment by zombieJ** [comment]
Checked. It's already fixed with latest `rc-table`. Please reinstall node_modueles instead.

ref: https://codesandbox.io/s/fixed-columns-and-header-ant-design-demo-j3ncg

**Comment by sb-lee** [comment]
Thanks~!!!!

**Comment by Cenume** [comment]
@zombieJ It seems to happen again on rc-table later than 7.3.x

**Comment by zombieJ** [comment]
@Cenume could you help to create a new issue for this?

**Comment by Nxtra** [comment]
I found a way to mitigate the issue, although it's not the solution for the base of the problem.

In state I keep a tableKey: `const [tableKey, setTableKey] = useState(uuidV4());`

I added the property `key={tableKey}` to my `Table` component.
```
<Table
  key={tableKey}
  data={data}
/>
```

Every time I when I know this issue will occur, I update this tableKey with an other uuid, which causes the table to rerender with the headers visible.
